src/models.jl

```diff
@@ -496,6 +473,320 @@ metadata_model(
+        info). If observation weights `w` are passed during machine construction then the 
+        weight assigned to each neighbor vote is the product of the kernel generated 
+        weight for that neighbor and the corresponding observation weight.
+     
+    """
+
+"""
+$(MMI.doc_header(KNNClassifier))
+
+KNNClassifier implements [K-Nearest Neighbors classifier](https://en.wikipedia.org/wiki/K-nearest_neighbor_algorithm) 
+which is non-parametric algorithm that predicts a discrete class distribution associated 
+with a new point by taking a vote over the classes of the k-nearest points. Each neighbor 
+vote is assigned a weight based on proximity of the neighbor point to the test point 
+according to a specified distance metric.
+
+For more information about the weighting kernels, see the paper by Geler et.al 
+[Comparison of different weighting schemes for the kNN classifier on time-series data](https://perun.pmf.uns.ac.rs/radovanovic/publications/2016-kais-knn-weighting.pdf). 
+
+# Training data
+In MLJ or MLJBase, bind an instance `model` to data with
+    mach = machine(model, X, y)
+OR
+    mach = machine(model, X, y, w)
+
+Here:
+
+- `X` is any table of input features (eg, a `DataFrame`) whose columns are of scitype
+  `Continuous`; check column scitypes with `schema(X)`.
+
+- `y` is the target, which can be any `AbstractVector` whose element scitype is
+  `<:Finite`; check the scitype with `scitype(y)`
+
+- `w` is the observation weights which can either be `nothing`(default) or an 
+  `AbstractVector` whoose element scitype is `Count` or `Continuous`. This is 
+  different from `weights` kernel which is an hyperparameter to the model, see below.
+
+Train the machine using `fit!(mach, rows=...)`.
+
+# Hyper-parameters
+
+$KNNFIELDS
+
+# Operations
+
+- `predict(mach, Xnew)`: Return predictions of the target given features `Xnew`, which
+  should have same scitype as `X` above. Predictions are probabilistic but uncalibrated.
+
+- `predict_mode(mach, Xnew)`: Return the modes of the probabilistic predictions
+  returned above.
+
+# Fitted parameters
+
+$KNNFITTEDPARAMS
+
+# Examples
+```
+using MLJ
+KNNClassifier = @load KNNClassifier pkg=NearestNeighborModels
+X, y = @load_crabs; # loads the crabs dataset from MLJBase
+# view possible kernels
+NearestNeighborModels.list_kernels()
+# KNNClassifier instantiation
+model = KNNClassifier(weights = NearestNeighborModels.Inverse())
+mach = machine(model, X, y) |> fit! # wrap model and required data in an MLJ machine and fit
+y_hat = predict(mach, X)
+labels = predict_mode(mach, X)
+
+```
+See also [`MultitargetKNNClassifier`](@ref)
+"""
+KNNClassifier
+
+"""
+$(MMI.doc_header(MultitargetKNNClassifier))
+
+Multi-target K-Nearest Neighbors Classifier (MultitargetKNNClassifier) is a variation of 
+[`KNNClassifier](@ref) that assumes the target variable is vector-valued with
+`Multiclass` or `OrderedFactor` components. 
```

Comment:
Probably we should mitigate possible confusion with requirements for `y` below. Something like:

```suggestion
`Multiclass` or `OrderedFactor` components. (Target data must be presented as a table, however.)
```